python/AMI_NOT_PUBLIC_CHECK/AMI_NOT_PUBLIC_CHECK.py

```diff
@@ -0,0 +1,398 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+'''
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+  AMI_NOT_PUBLIC_CHECK
+
+Description:
+  Check whether the Amazon Machine Images are not publicly accessible. The rule is NON_COMPLIANT if one or more Amazon Machine Images are publicly accessible.
+
+Trigger:
+  Periodic
+
+Reports on:
+  AWS::::Account
+
+Rule Parameters:
+  None
+
+Scenarios:
+  Scenario: 1
+    Given: No AMIs with "is-public" parameter set to True
+     Then: Return COMPLIANT
+  Scenario: 2
+    Given: One or more AMIs with is-public parameter set to True
+     Then: Return NON_COMPLIANT with Annotation containing AMIIDs
+'''
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+# Generates non-compliant response
+def generate_evaluation_list(images, event):
+    evaluations = []
+    for image in images:
+        evaluation = build_evaluation(
+            image['ImageId'],
+            "NON_COMPLIANT",
+            event,
+            annotation="Amazon Machine Image Id: {} is public.".format(image['ImageId'])
+        )
+        evaluations.append(evaluation)
```

Comment:
I didn't realize, but this doesn't follow the Gherkhin. You are creating several evaluations. But you shall create a unique evaluation on the AWS::::Account, with the Image IDs in the annotation. Please re-read scenario 2.
Be careful if the annotation go over 255 character it will not work. So you need to truncate it. Check with Rafih, we have the same conversation on EBS_SNAPSHOT_PUBLIC

Reply:
Done